components/Util/NumericInput.tsx

```diff
@@ -0,0 +1,40 @@
+export default function NumericInput({
+  label,
+  name,
+  max,
+  min,
+  step,
+  defaultValue,
+  value,
+  onChange,
+}: {
+  label: string
+  name?: string
+  max?: number
+  min?: number
+  step?: number
+  defaultValue?: number
+  value?: number
+  onChange?: (evt: any) => void
```

Comment:
Let's make it so this component returns a `number`. To me it makes sense to handle the event itself inside this component and only pass the resulting number up to the parent.  If I choose to use a `NumericInput` I expect the value that comes from it to be a number.

Reply:
Thanks for the review @afterburn.

I have some doubts to make about your comment.

> To me it makes sense to handle the event itself inside this component

Wouldn't handling the event inside the component make it an _uncontrolled component_? By definition, _controlled component_ states are governed by their parents.
I have allowed the possibility of both based on the description in #443 (For engineers, point 2).

Reply:
What I suggest is that we handle the event inside the component, this does not make it an uncontrolled component because as a developer you can still choose to manage the state for the input in the parent or have the `NumericInput` track state which is what the difference is between a controlled and uncontrolled component, where we handle the event has nothing to do with that.

Something like this:
```tsx
function NumericInput() {
  const [value, setValue] = useState(value)

  const handleChange = e => {
    const newValue = Number(e.target.value)
    
    // Based on whether its a controlled or uncontrolled component:
    if (controlled) {
       onChange(newValue)
    } else {
      setValue(newValue)
    }
  }

  return <input onChange={handleChange} />
}
```

This makes the component easier to use, which is the point:

```
function Parent() {
  const handleChange = (val: number) => {
    console.log(val) // Outputs number
  }
  
  return <NumericInput onChange={handleChange} />
}
```

Reply:
Full example of what I mean [here](https://codesandbox.io/s/numeric-input-xgpd6t)

Reply:
Thanks, @afterburn, for the example. This helps clears up my doubts!

Let me take references from it to create the final component!

Reply:
Addressed in 1c7cd6cb913ce1305fa02df9af0a4e5b9c955001
I have taken reference from your example to create the final component.
Thanks for the detailed example!